### ISSUE TITLE
Creating dynamic framework failed.

### ISSUE THREAD

**Issue by IamAlasca** [CocoaPods/cocoapods-packager#232]
Hi, I'm recently building a framework with a few dependencies in podspec file, one of them is like this "s.dependency 'Protobuf'". And i don't know why it keeps failing me with argument errors as below. I use the same script by changing the key words to create static library and it works totally fine. But adding "--dynamic" it's not working.  Any reasons why?  Many thanks.
### Report

* What did you do?

* What did you expect to happen?

* What happened instead?

### Stack

```
   CocoaPods : 1.6.1
        Ruby : ruby 2.3.7p456 (2018-03-28 revision 63024) [universal.x86_64-darwin17]
    RubyGems : 2.6.14
        Host : Mac OS X 10.13.6 (17G65)
       Xcode : 10.1 (10B61)
         Git : git version 2.17.2 (Apple Git-113)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ f47c108fd885d3d71e024fef71c1eea9db8139a2
               ximalaya-xmnetworkrequest - git@gitlab.ximalaya.com:iphone/XMNetworkRequest.git @ 290bb1b3b46a1ccf0da4b97d523e16654168e1a1
               ximalaya-xmspecs - git@gitlab.ximalaya.com:iphone/XMSpecs.git @ 499a0dedbed626ceabeae4abccb39b26e6fbbada
```

### Plugins

```
cocoapods-deintegrate : 1.0.3
cocoapods-packager    : 1.5.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
```

### Error

```
ArgumentError - wrong number of arguments (given 3, expected 7..9)
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.6.1/lib/cocoapods/target/pod_target.rb:64:in `initialize'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-packager-1.5.0/lib/cocoapods-packager/pod_utils.rb:146:in `new'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-packager-1.5.0/lib/cocoapods-packager/pod_utils.rb:146:in `build_dynamic_target'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-packager-1.5.0/lib/cocoapods-packager/pod_utils.rb:119:in `install_dynamic_pod'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-packager-1.5.0/lib/pod/command/package.rb:97:in `build_in_sandbox'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-packager-1.5.0/lib/pod/command/package.rb:113:in `block in build_package'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-packager-1.5.0/lib/pod/command/package.rb:112:in `each'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-packager-1.5.0/lib/pod/command/package.rb:112:in `build_package'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-packager-1.5.0/lib/pod/command/package.rb:80:in `run'
/Library/Ruby/Gems/2.3.0/gems/claide-1.0.2/lib/claide/command.rb:334:in `run'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.6.1/lib/cocoapods/command.rb:52:in `run'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.6.1/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:22:in `load'
/usr/local/bin/pod:22:in `<main>'
```

**Comment by hello-david** [comment]
+1. it happend to me, try using `pod _1.5.3_ package xxxx` may work

**Comment by amorde** [comment]
Fixed in #236
